gensim/models/base_any2vec.py

```diff
@@ -0,0 +1,792 @@
+
+    def _set_keyedvectors(self):
+        self.wv.vectors = self.trainables.__dict__.get('vectors', [])
+        self.wv.vectors_norm = self.trainables.__dict__.get('vectors_norm', None)
+        self.wv.vector_size = self.trainables.__dict__.get('vector_size', None)
+        self.wv.vocab = self.vocabulary.__dict__.get('vocab', {})
+        self.wv.index2word = self.vocabulary.__dict__.get('index2word', [])
+
+    def _log_progress(self, job_queue, progress_queue, cur_epoch, example_count, total_examples,
+                      raw_word_count, total_words, trained_word_count, elapsed):
+        if total_examples:
+            # examples-based progress %
+            logger.info(
+                "EPOCH %i - PROGRESS: at %.2f%% examples, %.0f words/s, in_qsize %i, out_qsize %i",
+                cur_epoch + 1, 100.0 * example_count / total_examples, trained_word_count / elapsed,
+                utils.qsize(job_queue), utils.qsize(progress_queue)
+            )
+        else:
+            # words-based progress %
+            logger.info(
+                "EPOCH %i - PROGRESS: at %.2f%% words, %.0f words/s, in_qsize %i, out_qsize %i",
+                cur_epoch + 1, 100.0 * raw_word_count / total_words, trained_word_count / elapsed,
+                utils.qsize(job_queue), utils.qsize(progress_queue)
+            )
+
+    def _log_epoch_end(self, cur_epoch, example_count, total_examples, raw_word_count, total_words,
+                       trained_word_count, elapsed):
+        logger.info(
+            "EPOCH - %i : training on %i raw words (%i effective words) took %.1fs, %.0f effective words/s",
+            cur_epoch + 1, raw_word_count, trained_word_count, elapsed, trained_word_count / elapsed
+        )
+
+        # check that the input corpus hasn't changed during iteration
+        if total_examples and total_examples != example_count:
+            logger.warning(
+                "EPOCH - %i : supplied example count (%i) did not equal expected count (%i)", cur_epoch + 1,
+                example_count, total_examples
+            )
+        if total_words and total_words != raw_word_count:
+            logger.warning(
+                "EPOCH - %i : supplied raw word count (%i) did not equal expected count (%i)", cur_epoch + 1,
+                raw_word_count, total_words
+            )
+
+    def _log_train_end(self, raw_word_count, trained_word_count, total_elapsed, job_tally):
+        logger.info(
+            "training on a %i raw words (%i effective words) took %.1fs, %.0f effective words/s",
+            raw_word_count, trained_word_count, total_elapsed, trained_word_count / total_elapsed
+        )
+        if job_tally < 10 * self.workers:
+            logger.warning(
+                "under 10 jobs per worker: consider setting a smaller `batch_words' for smoother alpha decay"
+            )
+
+    # for backward compatibility
+    @deprecated("Method will be removed in 4.0.0, use self.wv.most_similar() instead")
```

Comment:
Next version will be `4.0.0`, feel free to remove this stuff.
CC: @piskvorky

Reply:
We will have to decide whether to remove the `deprecated` methods in this PR. It is not as simple as just removing the `deprecated` methods from here. We will also have to update this changes in tutorials, notebooks and unittest cases -- as they use calls to these deprecated methods (`model.most_similar` is used almost everywhere instead of `model.wv.most_similar`).

Reply:
Aha, OK, let's do it in different PR later.

Reply:
But I'm confused yet because you already remove some deprecated stuff here, this is OK?
CC: @janpom